src/core/components/layout-utils/drop-down.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { PureComponent } from "react"
+import PropTypes from "prop-types"
+import cx from "classnames"
+
+export default class DropDown extends PureComponent {
+
+  static propTypes = {
+    id: PropTypes.string,
+    label: PropTypes.string,
+    displayLabel: PropTypes.string,
+    mod: PropTypes.string,
+    disbaled: PropTypes.bool,
+    children: PropTypes.node.isRequired
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      expanded: false
+    }
+  }
+
+  onClick = () => {
+    this.setState((state) => ({ expanded: !state.expanded }))
+  }
+
+  openDropdown = () => {
+    this.setState(() => ({ expanded: true }))
+    // set focus to first item/selected item
+  }
+
+  closeDropdown = () => {
+    this.setState(() => ({ expanded: false }))
+    // set focus to button
+  }
+
+  onKeyPress = (e) => {
+    e.preventDefault()
+    switch (e.key)
+    {
+      case "ArrowUp":
+        // if closed - open
+        // else - move focus to prev item
+        break
+      case "ArrowDown":
+        // if closed - open
+        // else - move focus to next item
+        break
+      case "Enter":
+      case " ":
+        this.openDropdown()
+        // if closed - open
+        // else - select item and close
+        break
+      case "Escape":
+        this.closeDropdown()
+        break
+      case "Home":
+        // if open - move to start of list
+        break
+      case "End":
+        // if open - move to end of list
+        break
+    }
+  }
+
+  displayLabel = () => (this.props.label && this.props.displayLabel)
+    ? <lable>{this.props.label}</lable>
+    : null
+
+  labelAttr = (el) => this.props.label
```

Comment:
`el` could be confusing here as it is normally used as a param that accepts a html node. Maybe `type` is more appropriate